server.go

```diff
@@ -344,14 +348,36 @@ func (s *Server) Serve(lis net.Listener) error {
 		}
 		s.mu.Unlock()
 	}()
+
+	var tempDelay time.Duration // how long to sleep on accept failure
+
 	for {
 		rawConn, err := lis.Accept()
 		if err != nil {
+			if ne, ok := err.(net.Error); ok && ne.Temporary() {
```

Comment:
This is not necessary to be `net.Error`. It could be any error which implements Temporary() interface. https://github.com/grpc/grpc-go/blob/ac7efbd8beb843cdb95d4bd275f0482729c6ad59/transport/http2_client.go#L130 can serve as an example.